posthog/api/test/test_survey.py

```diff
@@ -2346,6 +2346,18 @@ def test_cannot_reduce_iterations_lt_current_iteration(self):
         assert response.status_code == status.HTTP_400_BAD_REQUEST
         assert response.json()["detail"] == "Cannot change survey recurrence to 1, should be at least 2"
 
+    def test_guards_for_nil_iteration_count(self):
+        survey = self._create_recurring_survey()
```

Comment:
this creates a survey with an iteration count, though, right?  Shouldn't our test handle the case where we don't have an iteration count and we confirm that the API doesn't fail, since that was the case that failed in production?